dask/optimize.py

```diff
@@ -427,6 +428,179 @@ def fuse_getitem(dsk, func, place):
                            lambda a, b: tuple(b[:place]) + (a[2], ) + tuple(b[place + 1:]))
 
 
+def fuse_reductions(dsk, keys=None, ave_width=2, max_depth_new_edges=None,
+                    max_height=None, max_width=None):
```

Comment:
We'll want ways to specify these from `dask.set_options` and `compute` calls.  I think the convention we started to use was to prepend the name of the optimization like the following:

```python
x.compute(fuse_ave_width=5)
```

Although adding all of these to the various `compute` methods seems tedious.  Perhaps check for the presence of `dask.context._globals['fuse_ave_width']` instead so that we can do the following:

```python
with dask.set_options(fuse_ave_width=5):
    x.compute()
```